### PR TITLE
LRDOCS-3968 npm build workflow docs

### PR DESCRIPTION
**Pull request by mwilliams2014**
@sez11a  here are the npm build workflow docs. Please review as soon as you can. Thx!

### CODE REVIEW

File: develop/tutorials/articles/110-portlets/15-javascript/04-npm/04-structure-of-osgi-bundles-npm.markdown [hunk 1]
```diff
@@ -6,43 +6,43 @@ and modify them to work with @product@'s [Liferay AMD Loader](https://github.com
...
-                        isobject: 2.1.0
-                    ...
-                - `lib/`
-                    -`index.js`
-                    ...
+- `my-bundle/`
+    - `META-INF/`
+        - `resources/`
+            - `package.json`
+                name: my-bundle-package
```
Comment (mwilliams2014) [inline develop/tutorials/articles/110-portlets/15-javascript/04-npm/04-structure-of-osgi-bundles-npm.markdown]: @sez11a I have a small formatting issue with the contents of the package.json file in the folder structure. Do you know how to handle this? It's formatted the same way in each tutorial that references it.

**Comment by jbalsas** [comment]
Thanks a lot, @mwilliams2014!! 

@sez11a, we need this published by DevCon, so we can share it properly. Do you think we can meet that deadline? Thanks!

/cc @izaera

File: develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    convenience, you can use [Blade CLI](https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/installing-blade-cli) 
+    to create an empty portlet with the [mvc portlet blade template](https://dev.liferay.com/develop/reference/-/knowledge_base/7-0/using-the-mvc-portlet-template).
+
+3.  Run the following command to install the liferay-npm-bundler:
+
+        npm install --save-dev liferay-npm-bundler
+
+    +$$$
+
+    **Note:** Use npm from within your portlet's
```
Comment (izaera) [inline develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown]: This is "incorrect": you don't need to use npm from 'src/main/resources/.....'. You use it from your project's root folder, where the package.json file lives.

File: develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    typical web project.
+
+    $$$
+
+Now that you have the liferay-npm-bundler installed, you can add it to your 
+npm build process next.
+
+## Adding liferay-npm-bundler to Your Build Process [](id=adding-liferay-npm-bundler-to-your-build-process)
+
+After everything is transpiled (if necessary) to ECMAScript in your build
```
Comment (izaera) [inline develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown]: After 'everything is transpiled (if necessary) to ECMAScript' I would add 'and converted to AMD format'.

Comment (izaera) [reply develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown]: I mean, it is the responsibility of the developer (not of the bundler) to transform his own code and produce a portal-suitable Javascript.

Then bundler only takes care of code inside npm packages, but not of the code of the project.

**Comment by izaera** [comment]
I like the documentation. It's very well structured and explained. Good job 👏

File: develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown (hunk 3 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown]: Is this better?
After any JavaScript in your file is transpiled (if necessary) to ECMAScript in 
your build script and converted to AMD format, you must run liferay-npm-bundler 
in your `package.json` to pack the needed npm packages and transform them to AMD. 
This lets [Liferay AMD Loader](https://github.com/liferay/liferay-amd-loader) 
grab them from the Portal.

Comment (mwilliams2014) [reply develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown]: After any source code in your file is transpiled (if necessary) to ECMAScript in 
your build script and converted to AMD format for the [Liferay AMD Loader](/develop/tutorials/-/knowledge_base/7-0/liferay-amd-module-loader), you 
must run liferay-npm-bundler in your `package.json` to pack the needed npm 
packages and transform them to AMD. This lets Liferay AMD Loader grab the 
packages from the Portal.

**Comment by mwilliams2014** [comment]
@sez11a updated with Ivan's suggestions

**Pull request closed without merging**